bots/pr/src/main/java/org/openjdk/skara/bots/pr/ReviewersTracker.java

```diff
@@ -59,19 +60,30 @@ static String setReviewersMarker(int numReviewers, String role) {
                     break;
                 }
             } else {
-                updatedLimits.replace(r, updatedLimits.get(r) + remainingAdditional);
+                if (remainingAdditional > updatedLimits.get(r)) {
+                    // Set the number for the lower roles to remove.
+                    remainingRemovals = remainingAdditional - updatedLimits.get(r);
+                    // The new number of the reviewers should larger or equal than the requirement of the '.jcheck/conf' file.
+                    // Because the '.jcheck/conf' file means the minimal reviewer requirement.
+                    updatedLimits.replace(r, remainingAdditional);
+                }
                 break;
             }
         }
 
+        if (remainingRemovals == 0) {
+            // Improve performance. If remainingRemovals is 0, don't need to decrease the lower roles.
+            return updatedLimits;
+        }
+
         // Decrease lower roles (if any) to avoid increasing the total count above the requested
         Collections.reverse(roles);
-        var remainingRemovals = count;
         for (var r : roles) {
             if (!r.equals(role)) {
-                var removed = Math.max(0, updatedLimits.get(r) - remainingRemovals);
+                var originalVal = updatedLimits.get(r);
+                var removed = Math.max(0, originalVal - remainingRemovals);
                 updatedLimits.replace(r, removed);
-                remainingRemovals -= removed;
+                remainingRemovals -= (originalVal - updatedLimits.get(r));
```

Comment:
Maybe use the local variable `removed` instead of updatedLimits.get(r). I think that reads better.

Reply:
Fixed